src/Mod/PartDesign/Gui/ReferenceSelection.cpp

```diff
@@ -86,13 +99,24 @@ bool ReferenceSelection::allow(App::Document* pDoc, App::DocumentObject* pObj, c
             return false;
     }
 #endif
-    // Handle selection of geometry elements
-    if (Base::Tools::isNullOrEmpty(sSubName))
+    if (!sSubName || sSubName[0] == '\0'){
+        if (pObj->isDerivedFrom<Sketcher::SketchObject>()) {
+            bool allowed = type.testFlag(AllowSelection::SKETCH);
+            return allowed;

```

Comment:
```suggestion
            return type.testFlag(AllowSelection::SKETCH);
```
Do not declare a variable for a unique use but add a code comment if you find the result unclear.